src/backend/x11/mod.rs

```diff
@@ -349,8 +349,7 @@ pub struct X11Surface {
     format: DrmFourcc,
     width: u16,
     height: u16,
-    current: Option<Slot<BufferObject<()>>>,
-    next: Option<Slot<BufferObject<()>>>,
+    buffer: Option<Slot<BufferObject<()>>>,
```

Comment:
Would be great to merge these changes in the x11 backend as their own PR so I can implement multi-window handling in peace.

Reply:
Good call, I will cherry-pick them.

Reply:
https://github.com/Smithay/smithay/pull/425